libbeat/docs/processors-config.asciidoc

```diff
@@ -453,3 +454,67 @@ section.
 NOTE: If you define an empty list of fields under `include_fields`, then only
 the required fields, `@timestamp` and `type`, are exported.
 
+[[kubernetes]]
+=== kubernetes
+
+The `kubernetes` processor annotates each event with relevant metadata based on
+which kubernetes pod the event originated from. Each event is annotated with:
+
+* Pod Name
+* Namespace
+* Labels
+
+The `kuberentes` process has two basic building blocks which are:
+
+* Indexers
+* Matchers
+
+Indexers take in a pod's metadata and builds indices based on the pod metadata. For example,
+the `ip_port` indexer can take a kubernetes pod and index the pod metadata based on all
+`pod_ip:container_port` combinations.
+
+Matchers are used to build lookup keys with which indices can be queried. For example, the `fields`
```

Comment:
A little hard to follow because of the passive voice. Maybe say, "...build lookup keys for querying indices."